src/mon/OSDMonitor.cc

```diff
@@ -3122,9 +3123,9 @@ bool OSDMonitor::check_failure(utime_t now, int target_osd, failure_info_t& fi)
       auto reporter_loc = osdmap.crush->get_full_location(p->first);
       if (auto iter = reporter_loc.find(reporter_subtree_level);
           iter == reporter_loc.end()) {
-        reporters_by_subtree.insert("osd." + to_string(p->first));
+        reporters_by_subtree_addr.insert(osdmap.get_addr(p->first).get_addr_to_string());
```

Comment:
if "mon_osd_reporter_subtree_level" is "host", i think this problem will disappear.

Reply:
@tchaikov, if "mon_osd_reporter_subtree_level" is "host",this problem will appear also, It has been confirmed.

Reply:
but if "mon_osd_reporter_subtree_level" is "host", the reporters should be grouped by "host" and the reports from the same host should be considered sent from the same source, am i right?